### ISSUE TITLE
Idiosyncratic failure in `test_lineWidth_sf`

### ISSUE THREAD

**Issue by jaketmp** [phenomecentre/nPYc-Toolbox#12]
Failure occurs in approx 1% of runs:

```
======================================================================
ERROR: test_lineWidth_sf (test_utilities.test_utilities_linewidth)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jtpearce/Dropbox (Personal)/Development/nPYc-toolbox/Tests/test_utilities.py", line 1016, in test_lineWidth_sf
    calculatedLW = lineWidth(x, self.y, sf, [-5, 5], multiplicity='singlet')
  File "../nPYc/utilities/_lineWidth.py", line 29, in lineWidth
    fit = fitPeak(X, ppm, peakRange, multiplicity, parameters=parameters, maxLW=maxLW, estLW=estLW, shiftTollerance=shiftTollerance)
  File "../nPYc/utilities/_fitPeak.py", line 305, in fitPeak
    fit = peak.fit(spec, pars, x=localPPM)
  File "/Users/jtpearce/anaconda/lib/python3.6/site-packages/lmfit/model.py", line 736, in fit
    output.fit(data=data, weights=weights)
  File "/Users/jtpearce/anaconda/lib/python3.6/site-packages/lmfit/model.py", line 951, in fit
    _ret = self.minimize(method=self.method)
  File "/Users/jtpearce/anaconda/lib/python3.6/site-packages/lmfit/minimizer.py", line 1649, in minimize
    return function(**kwargs)
  File "/Users/jtpearce/anaconda/lib/python3.6/site-packages/lmfit/minimizer.py", line 1408, in leastsq
    eval_stderr(par, uvars, result.var_names, params)
  File "/Users/jtpearce/anaconda/lib/python3.6/site-packages/lmfit/minimizer.py", line 108, in eval_stderr
    uval = wrap_ueval(*uvars, _obj=obj, _names=_names, _pars=_pars)
  File "/Users/jtpearce/anaconda/lib/python3.6/site-packages/lmfit/uncertainties/__init__.py", line 696, in f_with_affine_output
    if arg.derivatives
  File "/Users/jtpearce/anaconda/lib/python3.6/site-packages/lmfit/uncertainties/__init__.py", line 492, in partial_derivative_of_f
    return (shifted_f_plus - shifted_f_minus)/2/step
ZeroDivisionError: float division by zero

----------------------------------------------------------------------
```

**Comment by Gscorreia89** [comment]
Haven't been able to reproduce this one yet, but looking on the code it seems to me the cause stems from random.random_sample() having 0 as a possible draw value. I have replaced the 
```
sf = (1000-200)*numpy.random.random_sample() + 200
```
with 
```
sf = numpy.random.uniform(200, 1000)
```
which should never return 0. Let's keep this one for a few more months, and if nobody sees it again we close it.

**Comment by jaketmp** [comment]
I've just run this ~800 times with no failures. I think you fixed it in ce403abffaaf2a74f7ff590713ce9347a1edfeac!
